src/main/java/com/ericsson/ei/frontend/utils/EIRequestsControllerUtils.java

```diff
@@ -104,31 +109,58 @@ private BackEndInformation getEIBackendInformation(HttpServletRequest request) {
         } catch (URISyntaxException e) {
             LOG.error("Error while encoding URL parameters: " + e);
         }
+
         return params;
     }
 
     private String extractUrlFromParameters(List<NameValuePair> params) {
         String urlFromParams = null;
+
         for (NameValuePair param : params) {
-            if (param.getName().equals(BACKEND_KEY_NAME)) {
-                urlFromParams = param.getValue();
+            if (param.getName()
+                     .equals(BACKEND_URL_KEY_NAME)) {
+                try {
+                    urlFromParams = URLDecoder.decode(param.getValue(), "UTF-8");
+                } catch (UnsupportedEncodingException e) {
+                    LOG.error("Failed to decode back-end url");
+                }
             }
         }
+
         return urlFromParams;
     }
 
-    private String removeBackendDataFromQueryString(List<NameValuePair> params) {
+    private String extractBackEndNameFromParameters(List<NameValuePair> params) {
+        String backendName = null;
+
+        for (NameValuePair param : params) {
+            if (param.getName()
+                     .equals(BACKEND_NAME_KEY_NAME)) {
+                try {
+                    backendName = URLDecoder.decode(param.getValue(), "UTF-8");
+                } catch (UnsupportedEncodingException e) {
+                    LOG.error("Failed to decode back-end name");
+                }
+            }
+        }
+
+        return backendName;
+    }
+
+    private String removeBackendDataFromQueryString(String requestQuery) {
+        List<NameValuePair> params = getParameters(requestQuery);
         List<NameValuePair> processedParams = new ArrayList<>();
         for (NameValuePair param : params) {
             String name = param.getName(), value = param.getValue();
-            if (name.equals(BACKEND_KEY_NAME)) {
+            if (name.equals(BACKEND_URL_KEY_NAME) || name.equals(BACKEND_NAME_KEY_NAME) || name == null || value == null) {
```

Comment:
put the name== null and value == null first. If name == null this would give exception anyways.

Reply:
Done